src/ray/gcs/redis_context.cc

```diff
@@ -184,6 +181,11 @@ void RedisRequestContext::RedisResponseFn(struct redisAsyncContext *async_contex
                    << "]"
                    << " failed due to error " << error_msg << ". "
                    << request_cxt->pending_retries_ << " retries left.";
+    // Reconnect if connection is lost or the error is a MOVED error.
+    if (redis_reply == nullptr ||
+        std::string(redis_reply->str).find("MOVED") != std::string::npos) {
```

Comment:
Does handling `MOVED` errors here mean that this would work with a multi-shard Ray Cluster? Does the GCS use multiple keys in Redis or just a single key?

Reply:
@JoshKarpel No. Please subscribe to https://github.com/ray-project/ray/issues/42919 for multi-shard.